### ISSUE TITLE
DeepSpeed & Lightning are both calling the scheduler

### ISSUE THREAD

**Issue by SeanNaren** [PyTorchLightning/pytorch-lightning#9771]
## 🐛 Bug

A user in our slack channel noticed that the scheduler step is being called by DeepSpeed and Lightning. Lightning should not call this, and DeepSpeed should be in control.

### To Reproduce

This test fails!
```python
import os

import mock
import torch
from torch.utils.data import DataLoader, Dataset

from pytorch_lightning import LightningModule, Trainer


class RandomDataset(Dataset):
    def __init__(self, size, length):
        self.len = length
        self.data = torch.randn(length, size)

    def __getitem__(self, index):
        return self.data[index]

    def __len__(self):
        return self.len


class BoringModel(LightningModule):
    def __init__(self):
        super().__init__()
        self.layer = torch.nn.Linear(32, 2)

    def forward(self, x):
        return self.layer(x)

    def training_step(self, batch, batch_idx):
        loss = self(batch).sum()
        self.log("train_loss", loss)
        return {"loss": loss}

    def validation_step(self, batch, batch_idx):
        loss = self(batch).sum()
        self.log("valid_loss", loss)

    def test_step(self, batch, batch_idx):
        loss = self(batch).sum()
        self.log("test_loss", loss)

    def configure_optimizers(self):
        optimizer = torch.optim.SGD(self.layer.parameters(), lr=0.1)
        scheduler = torch.optim.lr_scheduler.StepLR(optimizer, 1, gamma=0.1)
        return {
            "optimizer": optimizer,
            "lr_scheduler": {
                "scheduler": scheduler,
            },
        }


@mock.patch('torch.optim.lr_scheduler.StepLR.step', autospec=True)
def test_deepspeed(mock_step):
    train_data = DataLoader(RandomDataset(32, 64), batch_size=2)
    val_data = DataLoader(RandomDataset(32, 64), batch_size=2)

    model = BoringModel()
    trainer = Trainer(
        default_root_dir=os.getcwd(),
        limit_train_batches=1,
        limit_val_batches=0,
        num_sanity_val_steps=0,
        max_epochs=1,
        fast_dev_run=True,
        gpus=1,
        weights_summary=None,
        plugins='deepspeed'
    )
    trainer.fit(model, train_dataloaders=train_data, val_dataloaders=val_data)
    # once before training, and once during training.
    assert mock_step.call_count == 2
```

### Expected behaviour

Above should succeed. I also see there may be an issue with `test` also calling the step function internally in DeepSpeed, but thats for a different issue.

cc @tchaton 

### Environment

* CUDA:
        - GPU:
                - NVIDIA GeForce RTX 3090
        - available:         True
        - version:           11.1
* Packages:
        - numpy:             1.21.2
        - pyTorch_debug:     False
        - pyTorch_version:   1.9.1
        - pytorch-lightning: 1.5.0dev
        - tqdm:              4.49.0
* System:
        - OS:                Linux
        - architecture:
                - 64bit
                - ELF
        - processor:         x86_64
        - python:            3.8.8

**Comment by awaelchli** [comment]
Does does plugin need to wrap these schedulers in order to prevent the duplicate step call? Do you see another solution?

**Comment by SeanNaren** [comment]
@awaelchli what do you mean by wrap? I need to investigate what is happening internally, as I think from my initial glance we don't save the LR states outside of deepspeed, so I wonder if there is any point exposing the LR scheduler outside of deepspeed (in lightning)
